coldfront/core/project/views_/new_project_views/request_views.py

```diff
@@ -371,6 +381,48 @@ def show_new_pi_form_condition(self):
         cleaned_data = self.get_cleaned_data_for_step(step) or {}
         return cleaned_data.get('PI', None) is None
 
+    def show_pi_department_form_condition(self):
+        if not flag_enabled('USER_DEPARTMENTS_ENABLED'):
+            return False
+        extra_data = self.request.session \
+            ['wizard_savio_project_request_wizard']['extra_data']
+        has_entry_key = 'ldap_lookup_has_entry'
+        email_key = 'ldap_lookup_email'
+
+        # get email, fn, ln
+        email = fn = ln = None
+        step_name = 'new_pi'
+        step = str(self.step_numbers_by_form_name[step_name])
+        cleaned_data = self.get_cleaned_data_for_step(step) or {}
+        if cleaned_data:
+            email = cleaned_data['email']
+            fn = cleaned_data['first_name']
+            ln = cleaned_data['last_name']
+        else:
+            step_name = 'existing_pi'
+            step = str(self.step_numbers_by_form_name[step_name])
+            cleaned_data = self.get_cleaned_data_for_step(step) or {}
+            if cleaned_data.get('PI', None):
+                if UserDepartment.objects.filter(
+                        userprofile=cleaned_data['PI'].userprofile).exists():
+                    return False
+                email = cleaned_data['PI'].email
+                fn = cleaned_data['PI'].first_name
+                ln = cleaned_data['PI'].last_name
+        if not email:
+            return False
+
+        if has_entry_key in extra_data and email == extra_data[email_key]:
+            has_entry = extra_data[has_entry_key]
+        else:
+            entry = ldap_search_user(email, fn, ln)
```

Comment:
It seems like we're only ever using the `ldap_search_user` function to get the department number. Can it be reworked/renamed to just return that?

This way, the caller (this code) doesn't need to know that the return value has a `departmentNumber` attribute.